axelrod/player.py

```diff
@@ -122,6 +125,44 @@ def __init__(self):
         self.state_distribution = defaultdict(int)
         self.set_match_attributes()
 
+
+    def __eq__(self, other):
+        """
+        Test if two players are equal.
+        """
+        if self.__repr__() != other.__repr__():
+            return False
+        for attribute in set(list(self.__dict__.keys()) +
+                             list(other.__dict__.keys())):
+
+            value = getattr(self, attribute, None)
+            other_value = getattr(other, attribute, None)
+
+            if isinstance(value, np.ndarray):
+                if not (np.array_equal(value, other_value)):
+                    return False
+
+            elif isinstance(value, types.GeneratorType) or \
+                 isinstance(value, itertools.cycle):
+
+                # Split the original generator so it is not touched
+                generator, original_value = itertools.tee(value)
+                other_generator, original_other_value = itertools.tee(other_value)
+
+                setattr(self, attribute,
+                        (ele for ele in original_value))
+                setattr(other, attribute,
+                        (ele for ele in original_other_value))
+
+                if not (all(next(generator) == next(other_generator)
```

Comment:
Instead of
```python
generator, original_value = itertools.tee(value)
other_generator, original_other_value = itertools.tee(other_value)

setattr(self, attribute,
    (ele for ele in original_value))
 setattr(other, attribute,
    (ele for ele in original_other_value))
```
Couldn't you just do :
```python
generator, value = itertools.tee(value)
other_generator, other_value = itertools.tee(other_value)
```
?